### PR TITLE
essPath improve

### PR DESCRIPTION
**Pull request by Elorfin**

### CODE REVIEW

File: main/core/Resources/views/Workspace/layout.html.twig [hunk 1]
```diff
@@ -90,6 +90,28 @@
             window.Workspace.id = null;
         {% endif %}
     </script>
+
+    <script>
+        var postHeight = function postHeight() {
+            if (parent) {
+                console.log('child = ' + document.body.scrollHeight);
```
Comment (stefk) [inline main/core/Resources/views/Workspace/layout.html.twig]: This log (and the other at line 107) is probably not necessary.

File: main/core/Resources/views/Workspace/layout.html.twig [hunk 2]
```diff
@@ -90,6 +90,28 @@
...
+
+                var message = 'documentHeight:'+document.body.scrollHeight;
```
Comment (stefk) [inline main/core/Resources/views/Workspace/layout.html.twig]: The message can be any sort of data, so an object might be a better fit here: `{ documentHeight: document.body.scrollHeight }`

File: main/core/Resources/views/Workspace/layout.html.twig [hunk 3]
```diff
@@ -90,6 +90,28 @@
...
+
+                parent.postMessage(message, "*");
```
Comment (stefk) [inline main/core/Resources/views/Workspace/layout.html.twig]: According to [mdn](https://developer.mozilla.org/fr/docs/Web/API/Window/postMessage), there's a security risk in using `*` as the targetOrigin parameter. If this listener is used for resources embedded in iframes, couldn't we use `document.origin` instead?

**Comment by stefk** [comment]
There's commented code (js and less) and extra log statements in a several places, otherwise it looks good. But as it's a rather large pr with ui changes, it might be a good idea to start a docker instance to review it.

**Comment by Elorfin** [comment]
Sorry. This was not supposed to be a new PR.
Some work in progress that have been integrated in other PRs.

**Pull request closed without merging**